cdap-common/src/test/java/co/cask/cdap/io/SchemaTest.java

```diff
@@ -464,6 +464,21 @@ public void testUnionRecordReference() throws Exception {
     Assert.assertEquals(first, union.getUnionSchemas().get(1).getField("firsts").getSchema().getComponentSchema());
   }
 
+  @Test
+  public void testLogicalTypes() throws Exception {
+    Schema dateType = Schema.nullableOf(Schema.of(Schema.LogicalType.DATE));
```

Comment:
The test should include having Array, Map and Record that involves logical type.